### DISCUSSION TITLE
server-side equivalent of go-vcr ?

### DISCUSSION THREAD

**Discussion by marco-m-pix4d** [dnaeon/go-vcr#88] [General]
Hello Marin,

thanks for go-vcr, I found it very helpful!

I am wondering how to best test the following scenario.

With go-vcr, the SUT (system under test) is the HTTP client. Imagine now wanting to test the other side, the HTTP server, by recording and then replaying the interaction at the server side.

A bit in the spirit of https://docs.mitmproxy.org ("Replay the client-side of an HTTP conversation"), but in Go and usable as a package, not as an external tool.

Currently I have a misty idea in my mind of writing a reverse proxy, but I am wondering if you already thought of this problem or if you knew of existing libraries for this.

thanks!

**Comment by dnaeon** [comment]
Hey Marco,

I haven't really thought about it to be honest, although I do see the benefits of such a package.

I'd be happy to further discuss such an idea and how to potentially integrate it into `go-vcr`.

Thanks,
Marin

**Reply by marco-m-pix4d** [reply]
Thanks Marin. I'll come back to this discussion when I have my idea solidified a bit. It is not as simple as I thought at the beginning :-)

**Comment by marco-m** [comment]
This feature is now available in v4.
